mam/v2/mam/channel.c

```diff
@@ -91,41 +91,98 @@ retcode_t mam_channel_unwrap(spongos_t *const spongos, trits_t *const buffer,
 }
 
 size_t mam_channel_serialized_size(mam_channel_t const *const channel) {
-  return 0;
+  size_t mss_size = mss_stored_size(&channel->mss);
+  size_t endpoints_size = mam_endpoints_serialized_size(channel->endpoints);
+
+  return pb3_sizeof_ntrytes(MAM2_CHANNEL_ID_SIZE / 3) +       // id
+         pb3_sizeof_size_t(trits_size(channel->name)) +       // name size
+         pb3_sizeof_ntrytes(trits_size(channel->name) / 3) +  // name
+         pb3_sizeof_size_t(mss_size) +                        // mss size
+         pb3_sizeof_ntrytes(mss_size / 3) +                   // mss
+         pb3_sizeof_ntrytes(endpoints_size / 3);              // endpoints
 }
 
 retcode_t mam_channel_serialize(mam_channel_t const *const channel,
                                 trits_t *const buffer) {
   size_t mss_size = mss_stored_size(&channel->mss);
 
-  return pb3_sizeof_ntrytes(MAM2_CHANNEL_ID_SIZE / 3) +
-         pb3_sizeof_size_t(mss_size) + pb3_sizeof_ntrytes(mss_size / 3);
+  pb3_encode_ntrytes(trits_from_rep(MAM2_CHANNEL_ID_SIZE, channel->id),
+                     buffer);                              // id
+  pb3_encode_size_t(trits_size(channel->name), buffer);    // name size
+  pb3_encode_ntrytes(channel->name, buffer);               // name
+  pb3_encode_size_t(mss_size, buffer);                     // mss size
+  mss_save(&channel->mss, trits_take(*buffer, mss_size));  // mss
+  trits_advance(buffer, mss_size);
+  mam_endpoints_serialize(channel->endpoints, buffer);  // endpoints
+
+  return RC_OK;
 }
 
 retcode_t mam_channel_deserialize(trits_t *const buffer,
                                   mam_channel_t *const channel) {
-  size_t mss_size = mss_stored_size(&channel->mss);
+  retcode_t ret = RC_OK;
+  size_t size = 0;
+  mss_mt_height_t height = 0;
 
-  fprintf(stderr, "SERIALIZE\n");
-  pb3_encode_ntrytes(trits_from_rep(MAM2_CHANNEL_ID_SIZE, channel->id), buffer);
-  pb3_encode_size_t(mss_size, buffer);
-  // TODO: update mss_save interface
-  mss_save(&channel->mss, trits_take(*buffer, mss_size));
-  trits_advance(buffer, mss_size);
+  if ((ret = pb3_decode_ntrytes(
+           trits_from_rep(MAM2_CHANNEL_ID_SIZE, channel->id), buffer)) !=
+      RC_OK) {  // id
+    return ret;
+  }
 
-  return RC_OK;
+  if ((ret = pb3_decode_size_t(&size, buffer)) != RC_OK) {  // name size
+    return ret;
+  }
+  if (trits_is_null(channel->name = trits_alloc(size))) {
+    return RC_OOM;
+  }
+  if ((ret = pb3_decode_ntrytes(channel->name, buffer)) != RC_OK) {  // name
+    return ret;
+  }
+
+  if ((ret = pb3_decode_size_t(&size, buffer)) != RC_OK) {  // mss size
+    return ret;
+  }
+
+  MAM2_TRITS_DEF0(ts, MAM2_MSS_SKN_SIZE);
+  ts = MAM2_TRITS_INIT(ts, MAM2_MSS_SKN_SIZE);
+  trits_set_zero(ts);
+
+  trits_copy(trits_take(*buffer, 4), trits_take(ts, 4));
```

Comment:
We should have a Defined value for that skn part (height)

Reply:
Sure